gadgets/trace_capabilities/test/unit/trace_capabilities_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/gadgetrunner"
+)
+
+type ExpectedTraceCapabilitiesEvent struct {
+	Cap string `json:"cap"`
+}
+
+type testDef struct {
+	name                string
+	generateEvent       func() error
+	requestedPermission string
+}
+
+func TestTraceCapabilitiesGadget(t *testing.T) {
+	gadgettesting.InitUnitTest(t)
+	testCases := []testDef{
+		{
+			name: "unshare",
+			generateEvent: func() error {
+				return syscall.Unshare(syscall.CLONE_NEWUTS)
+			},
+			requestedPermission: "CAP_SYS_ADMIN",
+		},
+		{
+			name: "raw_socket",
+			generateEvent: func() error {
+				_, err := syscall.Socket(syscall.AF_INET, syscall.SOCK_RAW, syscall.IPPROTO_ICMP)
```

Comment:
If the syscall was successfull, the socket should be closed again

Reply:
Not only for this, all successful changes should be rolled back to leave the process on the same estate as before.